src/parser/symtable.ts

```diff
@@ -962,8 +963,8 @@ export class SymbolTable {
         this.cur.comp_iter_target = false;
 
         /* Visit the rest of the comprehension body */
-        this.SEQ(this.visitExpr, outermost.ifs);
-        this.SEQTail(this.visitComprehension, generators, 1);
+        this.SEQ(this.visitExpr.bind(this), outermost.ifs);
```

Comment:
this or `(e=>this.visitExpr(e))` <-- this creates a lot more closures `bind(this)` is a bit magical 🤷